src/runtime/GlobalObjectBuiltinRegExp.cpp

```diff
@@ -350,6 +350,139 @@ static Value builtinRegExpSplit(ExecutionState& state, Value thisValue, size_t a
     return A;
 }
 
+static Value builtinRegExpReplace(ExecutionState& state, Value thisValue, size_t argc, Value* argv, bool isNewExpression)
+{
+    ASSERT(argv != nullptr);
+    RESOLVE_THIS_BINDING_TO_OBJECT(obj, RegExp, replace);
+    RegExpObject* rx = obj->asRegExpObject(state);
```

Comment:
I think the next testcase would cause assertion fail:

```
RegExp.prototype[Symbol.replace].call({}).
```

This routine must be generic, so you cannot assume that `rx` is a RegExp object.